packages/mailchimp/src/Utils.js

```diff
@@ -8,3 +10,58 @@ export function handleResponse(response, state, callback) {
   if (callback) return callback(nextState);
   return nextState;
 }
+
+const createClient = baseUrl => new Client(baseUrl);
+
+const setHeaders = config => ({
+  'Content-Type': 'application/json',
+  Authorization: `Basic ${Buffer.from(
+    `openfn:${config.apiKey}`,
+    'utf-8'
+  ).toString('base64')}`,
+});
+
+const defaultOptions = {
+  query: {},
+  body: undefined,
+};
+
+export const request = (method, path, options, callback = s => s) => {
+  return async state => {
+    // TODO should we add apiVersion in credentials schema ? the default can be 3.0
+    const apiVersion = '3.0';
+    const { apiKey, server } = state.configuration;
+
+    const [resolvedMethod, resolvedPath, resolvedOptions] = expandReferences(
+      state,
+      method,
+      path,
+      options
+    );
+    const { query, body } = { ...defaultOptions, ...resolvedOptions };
+
+    const headers = setHeaders({ apiKey });
+
+    const baseUrl = `https://${server}.api.mailchimp.com`;
+
+    const client = createClient(baseUrl);
+
+    const response = await client.request({
+      method: resolvedMethod,
+      path: `/${apiVersion}${resolvedPath}`,
+      headers,
+      query,
+      body: body ? JSON.stringify(body) : undefined,
+    });
+
+    const responseBody = await response.body.json();
+
+    return handleResponse(responseBody, state, callback);
```

Comment:
Actually I don't think we want to use `handleResponse` here.

Looking at the implementation it'll write the body to both `data` and `response`, so you'lll get a state object which looks like this:
```
{
    data: [/* mailchimp data */],
    response: [/* mailchimp data */]
}
```

Reply:
I removed the handleResponse from `request()` function